mod_form.php

```diff
@@ -478,6 +563,39 @@ public function data_postprocessing($data) {
                 }
             }
         }
+
+        // Add some postprocessing around the recurrence settings.
+        if ($data->recurring) {
+            // If "No fixed time" meeting selected, dont need repeat_interval and other options.
+            if ($data->recurrence_type == ZOOM_RECURRINGTYPE_NOTIME) {
+                unset($data->repeat_interval);
+                // Unset the weekly fields.
+                $data = zoom_remove_weekly_options($data);
+                // Unset the monthly fields.
+                $data = zoom_remove_monthly_options($data);
+                // Unset end_times and end_date.
+                unset($data->end_date_option);
+                unset($data->end_times);
+                unset($data->end_date_time);
+            }
+            // If daily recurring selected, unset weekly and monthly options.
+            if ($data->recurrence_type == ZOOM_RECURRINGTYPE_DAILY) {
+                // Unset the weekly fields.
+                $data = zoom_remove_weekly_options($data);
+                // Unset the monthly fields.
+                $data = zoom_remove_monthly_options($data);
+            }
+            // If weekly recurring selected, unset monthly options.
+            if ($data->recurrence_type == ZOOM_RECURRINGTYPE_WEEKLY) {
+                // Unset the monthly fields.
+                $data = zoom_remove_monthly_options($data);
+            }
+            // If daily recurring selected, unset weekly and monthly options.
+            if ($data->recurrence_type == ZOOM_RECURRINGTYPE_MONTHLY) {
+                // Unset the weekly fields.
+                $data = zoom_remove_weekly_options($data);
+            }
```

Comment:
I read this through a couple times and I think it has the same effect but with less duplication.
```suggestion
            // If "No fixed time" meeting selected, don't need repeat_interval and other options.
            if ($data->recurrence_type == ZOOM_RECURRINGTYPE_NOTIME) {
                unset($data->repeat_interval);
                // Unset end_times and end_date.
                unset($data->end_date_option);
                unset($data->end_times);
                unset($data->end_date_time);
            }
            // If weekly recurring is not selected, unset weekly options.
            if ($data->recurrence_type != ZOOM_RECURRINGTYPE_WEEKLY) {
                $data = zoom_remove_weekly_options($data);
            }
            // If monthly recurring is not selected, unset monthly options.
            if ($data->recurrence_type != ZOOM_RECURRINGTYPE_MONTHLY) {
                $data = zoom_remove_monthly_options($data);
            }
```

Reply:
thanks for the suggestion, have refactored this chunk of code